src/common/src/telemetry/report.rs

```diff
@@ -91,9 +100,30 @@ where
                 )
             });
 
+        let (tx, mut event_rx) = tokio::sync::mpsc::unbounded_channel::<PbEventMessage>();
+        TELEMETRY_EVENT_REPORT_TX.set(tx).unwrap_or_else(|_| {
+            tracing::warn!(
+                "Telemetry failed to set event reporting tx, event reporting will be disabled"
+            );
+        });
+        let mut event_stash = Vec::new();
+
         loop {
             tokio::select! {
                 _ = interval.tick() => {},
+                event = event_rx.recv() => {
+                    if let Some(event) = event {
```

Comment:
Given that `tx` is held in a static var, which will never be dropped, `recv()` should never return `None`. To be safer, can we assert that in debug build (cfg!(debug_assertions) == true) and print a warn log in release build?